widgets/linechart/linechart.go

```diff
@@ -232,6 +232,13 @@ func (lc *LineChart) Series(label string, values []float64, opts ...SeriesOption
 	return nil
 }
 
+func (lc *LineChart) Reset() {
```

Comment:
Thank you for the contribution. We can certainly add a `Reset` method on the `LineChart` widget, but there might be a more elegant solution. If we add it here, we will have to solve this problem on each widget. An alternative is to use the dynamic layout feature in the `container` layer.

With dynamic layout, we can tag a `container` with an arbitrary `ID` and later use that `ID` to reference the `container` and replace its `widget`:

https://github.com/mum4k/termdash/wiki/Dynamic-layout#replacing-widget-in-a-container

So to achieve the desired functionality you can:
- set an `ID` on the target `container` when creating it.
- create a new and empty instance of the `LineChart` widget.
- use the container's `Update` method together with the `ID` and the `container.PlaceWidget` option as in the example linked above.

If you prefer to continue adding the `Reset` method here, we need to do a bit more work:
- make the `Reset` method thread safe by acquiring the mutex on the `LineChart` object.
- reset all the fields, not just `series` (might be simpler to just replace self with a new object).
- document the method.
- add test coverage for the method.

Please let me know how you would like to proceed.

Reply:
Hi, @mum4k i think that would workable, but if i have some widgets in a container, like this:
![image](https://github.com/mum4k/termdash/assets/9976943/197b023d-256d-46b1-a647-567fbecf372b)
If i just clean only the LineChart data, do i need to do this again to output the new gridOpts?
```
builder.Add(elementFromGraphAndLegend(gr.widgetExprInput, gr.widgetExprLegend, gr.widgetGraph, gr.widgetGraphLegend))
gridOpts, err := builder.Build()
```

Reply:
The same approach works with the Grid layout. The idea is:

1. Mark the container where we want to replace the widget with an ID we know.
2. Update the container when required and use the `container.PlaceWidget` option to replace the old widget with any new widget.

Here is [an example](https://github.com/mum4k/termdash/pull/379/files) of how to achieve `Linechart` widget replacement with the grid layout. Feel free to run the demo and press `r` to replace the main linechart with another one.